### PR TITLE
Added __str__ method to classes.

### PR DESCRIPTION
**Pull request by shreyanshrs44**
I have added __str__ method to classes in /ogr/services/github.py , /ogr/services/gitlab.py,  /ogr/services/pagure.py.
Please have a look and give feedback on necessary changes.
I have not touched the classes inside /ogr/abstract.py as it containes __str__ method for most of classes in it.

### CODE REVIEW

File: ogr/services/github.py [hunk 1]
```diff
@@ -102,6 +106,13 @@ def __init__(
             )
         self.read_only = read_only
 
+    def __str__(self) -> str:
+        return (
+            f"Github-Project"
+            f"namespace: '{self.namespace}'"
+            f"repo: '{self.repo}'"
```
Comment (TomasTomecek) [inline ogr/services/github.py]: this won't look pretty, please wrap the parameters in brackets: `GithubProject(namespace: 'asd', ...)` and so on

File: ogr/services/github.py [hunk 2]
```diff
@@ -528,6 +539,10 @@ class GithubUser(BaseGitUser):
     def __init__(self, service: GithubService) -> None:
         super().__init__(service=service)
 
+    def __str__(self) -> str:
+        return (
+            f"Github-User"
```
Comment (TomasTomecek) [inline ogr/services/github.py]: would be nice to see the login

**Comment by lachmanfrantisek** [comment]
[test]

File: ogr/services/github.py (hunk 1 above)
Comment (shreyanshrs44) [reply ogr/services/github.py]: Yes, I will make the changes. Thanks for feedback !

File: ogr/services/github.py (hunk 2 above)
Comment (shreyanshrs44) [reply ogr/services/github.py]: Yes I will make the changes. Thanks for feedback !

File: ogr/services/github.py [hunk 3]
```diff
@@ -102,6 +106,13 @@ def __init__(
             )
         self.read_only = read_only
 
+    def __str__(self) -> str:
+        return (
+            f"Github-Project("
+            f"namespace: '{self.namespace}'"
+            f"repo: '{self.repo}')"
+        )
```
Comment (lachmanfrantisek) [inline ogr/services/github.py]: Personally, I like the instantiation-like syntax. You can sometimes use it to "recreate" the original object:

```python
>>> str(GithubProject(repo="my_repo", namespace="my_namespace"))
'GithubProject(repo="my_repo", namespace="my_namespace")'
```

Comment (shreyanshrs44) [reply ogr/services/github.py]: Yeah, I will make required changes.
Thanks for your feedback.

File: ogr/services/github.py [hunk 4]
```diff
@@ -102,6 +106,13 @@ def __init__(
             )
         self.read_only = read_only
 
+    def __str__(self) -> str:
+        return (
+            f"Github-Project("
+            f"namespace= \"{self.namespace}\""
+            f"repo: \"{self.repo}\")"
```
Comment (lachmanfrantisek) [inline ogr/services/github.py]: You've forgot to  repo parameter..;-)

```suggestion
            f"repo:=\"{self.repo}\")"
```

File: ogr/services/pagure.py [hunk 5]
```diff
@@ -198,7 +202,7 @@ def __init__(
         self.namespace = namespace
 
     def __str__(self) -> str:
-        return f"namespace={self.namespace} repo={self.repo}"
+        return f"namespace=\"{self.namespace}\" repo=\"{self.repo}\""
```
Comment (lachmanfrantisek) [inline ogr/services/pagure.py]: Please, be consistent and do it in a same way as you did in the case of Github.

```suggestion
        return f"GithubProject(namespace=\"{self.namespace}\", repo=\"{self.repo}\")"
```

File: ogr/services/pagure.py [hunk 6]
```diff
@@ -50,6 +50,10 @@ def __init__(
 
         self.header = {"Authorization": "token " + self._token} if self._token else {}
 
+    def __str__(self) -> str:
+        return (
+            f"Pagure-Service"
```
Comment (lachmanfrantisek) [inline ogr/services/pagure.py]: What about adding also `instance_url`. It would be useful to see it...

File: ogr/services/gitlab.py [hunk 7]
```diff
@@ -26,6 +26,11 @@ def __init__(self, token=None, url=None, full_repo_name=None):
         if full_repo_name:
             self.repo = self.g.projects.get(full_repo_name)
 
+    def __str__(self) -> str:
+        return(
+            f"GitLab Service"
```
Comment (lachmanfrantisek) [inline ogr/services/gitlab.py]: The GitLab currently does not work. We can work on the `__str__` after it is ready.

File: ogr/services/github.py [hunk 8]
```diff
@@ -55,6 +55,10 @@ def __init__(
         self.github = self.github_class(login_or_token=self._token)
         self.read_only = read_only
 
+    def __str__(self)-> str:
+        return (
+            f"Github-Service"
```
Comment (lachmanfrantisek) [inline ogr/services/github.py]: Please, use the same names as in code:
```suggestion
            f"GithubService"
```
It is relevant for other methods as well... For classes as well for attributes.

-----

What about adding `read_only` attribute. (You can add it also to the `PagureService`.)

File: ogr/services/pagure.py [hunk 9]
```diff
@@ -618,6 +622,10 @@ class PagureUser(BaseGitUser):
     def __init__(self, service: PagureService) -> None:
         super().__init__(service=service)
 
+    def __str__(self) -> str:
+        return (
+            f"Pagure-User"
```
Comment (lachmanfrantisek) [inline ogr/services/pagure.py]: Same as the previous comment. Try to have same info/format for both implementations.
```suggestion
            f'PagureUser(username="{self.get_username()}")'
```

File: ogr/services/github.py (hunk 4 above)
Comment (shreyanshrs44) [reply ogr/services/github.py]: My bad :(. I will make the changes. Thanks for your valuable feedback !

File: ogr/services/pagure.py (hunk 5 above)
Comment (shreyanshrs44) [reply ogr/services/pagure.py]: Makes sense. I will make the changes. Thanks for your valuable feedback !

File: ogr/services/pagure.py (hunk 6 above)
Comment (shreyanshrs44) [reply ogr/services/pagure.py]: Yes. It will be. I will make the changes. Thanks for your valuable feedback !

File: ogr/services/gitlab.py (hunk 7 above)
Comment (shreyanshrs44) [reply ogr/services/gitlab.py]: Should I leave as it is for now ? or should I remove the method ?

Comment (lachmanfrantisek) [reply ogr/services/gitlab.py]: You can remove it. The attributes may change before we make this instance work...

File: ogr/services/pagure.py (hunk 9 above)
Comment (shreyanshrs44) [reply ogr/services/pagure.py]: Yes, I will make the requested changes. Thanks for valuable feedback !

File: ogr/services/gitlab.py (hunk 7 above)
Comment (shreyanshrs44) [reply ogr/services/gitlab.py]: Yes, I will do it.

File: ogr/services/pagure.py (hunk 9 above)
Comment (lachmanfrantisek) [reply ogr/services/pagure.py]: No problem. Thanks for your work!

**Comment by shreyanshrs44** [comment]
Yes, I don't have any problem. Thanks for your valuable feedback in every review. It helped a lot !

**Comment by lachmanfrantisek** [comment]
> Yes, I don't have any problem. Thanks for your valuable feedback in every review. It helped a lot !

No problem. Just look at the tools we use (the links in the previous review) if you would like to contribute more...

They're useful in general.

**Comment by shreyanshrs44** [comment]
Yes, I will do.

**Comment by lachmanfrantisek** [comment]
@shreyanshrs44 Since you used the master branch as a source, I cannot edit your branch.

I've created a new PR (#98 ) with your commits and one mine on top of it. Thank you for your work!

**Pull request closed without merging**